File: pandorabox_proxy.rb
require 'sinatra'
require 'httparty'
require 'json'

get '/' do
  content_type :json
  headers \
    "Access-Control-Allow-Origin" => "*"

  options = { :query => { :botid => params[:botid], :input => params[:message], :custid => params[:custid] } }
  response = HTTParty.post('http://www.pandorabots.com/pandora/talk-xml', options)
  $stderr.puts "Response:"
  $stderr.puts response.to_yaml
  response.parsed_response.to_json
end

File: pandorabox_proxy.js
var http = require('http');

http.createServer(function (request, response) {
  response.writeHead(200, {
    'Content-Type': 'application/json',
    'Access-Control-Allow-Origin' : '*'
  });

  console.log('Request\n');

  response.end('Hello World\n');
}).listen(8124, '127.0.0.1');

console.log('Server running at http://127.0.0.1:8124/');
